docker/entrypoints/entrypoint_fix_permissions.sh

```diff
@@ -7,3 +7,85 @@ if [ ! -z "$USER_ID" ] && [ ! -z "$GROUP_ID" ]; then
     chown -R beetle:beetle /repo
     chown -R beetle:beetle /logs
 fi
+
+# Add extra groups to the beetle user if EXTRA_GROUPS is set
+# Format: "group_name1:gid1,group_name2:gid2" e.g., "nas_shares:1001,media:1002"
+if [ ! -z "$EXTRA_GROUPS" ]; then
+    echo "[Entrypoint] Adding extra groups to beetle user: $EXTRA_GROUPS"
+    # Split by comma and iterate
+    OLD_IFS="$IFS"
+    IFS=','
+    for group_spec in $EXTRA_GROUPS; do
```

Comment:
Can we refactor this into functions, one for the outer loop and one for processing/adding each group? Also, it would be great to remove the continue statements, as the current control flow is a bit hard to follow.